src/contracts/GPv2Settlement.sol

```diff
@@ -142,4 +150,57 @@ contract GPv2Settlement {
         );
         return IUniswapV2Pair(uint256(pairAddressBytes));
     }
+
+    /// @dev Verifies the specified clearing price is valid. Spefically, this
+    /// method verifies that the ±0.1% fee spread around the clearing price is
+    /// completely within the ±0.3% AMM fee spread around the Uniswap AMM price.
+    /// The AMM price is either defined by the token reserves (AKA spot price)
+    /// in the case the AMM is not traded with, or the actual swap amount (AKA
+    /// effective price) otherwise.
+    function verifyClearingPrice(
+        IUniswapV2Pair pair,
+        int112 d0,
+        int112 d1,
+        uint112 clearingPrice0,
+        uint112 clearingPrice1
```

Comment:
I'm having a bit of a hard time with these variable names here. E.g. it's not clear to me if a negative d0 means token 0 is being taken out of the AMM or passed in. 
Also clearingPrice0/1 would maybe nicer to express as numerator denominator for the price pair.0/pair.1 (although I might be the only one that thinks so).

Reply:
> I'm having a bit of a hard time with these variable names here. E.g. it's not clear to me if a negative d0 means token 0 is being taken out of the AMM or passed in.

`d*` are the AMM rebalancing amounts, so the change to the reserves. So a negative `d*` indicates that that token is being removed from the reserves, i.e. taken out of the AMM.

>  Also clearingPrice0/1 would maybe nicer to express as numerator denominator for the price pair.0/pair.1 (although I might be the only one that thinks so).

I had numerator/denominator originally, but I also thought this was confusing as it implies the price to be be quoted in a specific token. Using `clearingPrice*` was easier for me as `clearingPrice0` is clearly a `token0` amount and `clearingPrice1` is a `token1` amount, which makes it easier to reason about how to apply the prices to amounts. For example if I have a `buyAmount0` how do I compute the equivalent `sellAmount1`? To me with numerator/denominator, its not immediately obvious, but with `price*`, its a bit easier: `amoun0 * price1 / price0 -> amount1`.

Reply:
> I'm having a bit of a hard time with these variable names here. E.g. it's not clear to me if a negative d0 means token 0 is being taken out of the AMM or passed in.

We're going to change this. See discussion in https://github.com/gnosis/oba-contracts/pull/107#discussion_r512947426 on encoding the trade direction with the token ordering.